crates/ol/block-assembly/src/error.rs

```diff
@@ -0,0 +1,22 @@
+//! Error types for block assembly operations.
+
+use strata_db::errors::DbError;
+use strata_ol_mempool::MempoolError;
+
+/// Errors that can occur during block assembly operations.
+#[derive(Debug, thiserror::Error)]
+pub enum BlockAssemblyError {
+    /// Database operation failed.
+    #[error("Database error: {0}")]
+    Database(#[from] DbError),
+
+    /// Mempool operation failed.
+    #[error("Mempool error: {0}")]
+    Mempool(#[from] MempoolError),
+
+    /// Invalid L1 block range where `from_block` height > `to_block` height.
+    #[error(
+        "Invalid L1 block range: from_block height ({from_height}) > to_block height ({to_height})"
```

Comment:
This error string seems a little hard to read imo, would be cleaner as

`invalid L1 block height range (from {from_height} to {to_height})`